packages/platforms/react-native/lib/backgrounding-listener.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  type BackgroundingListener,
+  type BackgroundingListenerCallback,
+  type BackgroundingListenerState
+} from '@bugsnag/core-performance'
+import { type AppStateStatic, type AppStateStatus } from 'react-native'
+
+export default function createBrowserBackgroundingListener (appState: AppStateStatic) {
+  const callbacks: BackgroundingListenerCallback[] = []
+  let state: BackgroundingListenerState =
+  // on iOS the app state may be 'unknown' on launch, so we treat this as 'in-foreground'
+    appState.currentState === 'active' || appState.currentState === 'unknown'
+      ? 'in-foreground'
+      : 'in-background'
+
+  const backgroundingListener: BackgroundingListener = {
+    onStateChange (backgroundingListenerCallback: BackgroundingListenerCallback): void {
+      callbacks.push(backgroundingListenerCallback)
+
+      // trigger the callback immediately if the app is already in the background
+      if (state === 'in-background') {
+        backgroundingListenerCallback(state)
+      }
+    }
+  }
+
+  const backgroundStateChanged = (newState: BackgroundingListenerState) => {
+    if (state === newState) return
+
+    state = newState
+    for (const callback of callbacks) {
+      callback(state)
+    }
+  }
+
+  appState.addEventListener('change', (state: AppStateStatus) => {
+    const newState = state === 'active'
```

Comment:
does this need to be `'unknown'` as well?

Reply:
I think `unknown` is an initial state, so shouldn't be sent in an event, but it can't hurt 👍🏾